File: API_DBFruits/API_DBFruits/Models/DanhMucSanPham.cs
using System;
using System.Collections.Generic;

namespace API_DBFruits.Models;

public partial class DanhMucSanPham
{
    public int DanhMucId { get; set; }

    public string TenDanhMuc { get; set; } = null!;

    public string? MoTa { get; set; }

    public virtual ICollection<SanPham> SanPhams { get; set; } = new List<SanPham>();
}

File: lib/models/DanhMucSanPham.dart
class DanhMucSanPham {
  DanhMucSanPham({
    required this.danhMucId,
    required this.tenDanhMuc,
    required this.moTa,
    required this.sanPhams,
  });

  final int? danhMucId;
  final String? tenDanhMuc;
  final String? moTa;
  final List<dynamic> sanPhams;

  factory DanhMucSanPham.fromJson(Map<String, dynamic> json) {
    return DanhMucSanPham(
      danhMucId: json["danhMucId"],
      tenDanhMuc: json["tenDanhMuc"],
      moTa: json["moTa"],
      sanPhams: json["sanPhams"] == null
          ? []
          : List<dynamic>.from(json["sanPhams"]!.map((x) => x)),
    );
  }
}
